src/Controller/Api/BookController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+namespace App\Controller\Api;
+
+use App\Entity\Book;
+use App\Service\BookManager;
+use FOS\RestBundle\Controller\AbstractFOSRestController;
+use FOS\RestBundle\Controller\Annotations as Rest;
+use FOS\RestBundle\Controller\Annotations\QueryParam;
+use FOS\RestBundle\Request\ParamFetcherInterface;
+use FOS\RestBundle\View\View;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+
+/**
+ * Class BookController
+ * @package App\Controller\Api
+ *
+ */
+class BookController extends AbstractFOSRestController
+{
+    private BookManager $bookManager;
+
+    public function __construct( BookManager $bookManager)
+    {
+        $this->bookManager = $bookManager;
+    }
+
+    /**
+     * @Rest\Get(path="/api/books")
+     * @Rest\View(serializerGroups={"Default"})
+     * @QueryParam(name="page", requirements="\d+", default="1", description="The page number")
+     * @QueryParam(name="name", default="", description="The name of the book")
+     * @QueryParam(name="genre", default="", description="The genre of the book")
+     *
+     * @param ParamFetcherInterface $paramFetcher
+     * @return View
+     */
+    public function getBooks(ParamFetcherInterface $paramFetcher): View
+    {
+        $pagination = $this->bookManager->getBooks($paramFetcher);
+
+        return $this->view($pagination, Response::HTTP_OK);
+    }
+
+
+    /**
+     * @Rest\Get(path="/api/books/{id}")
+     * @Rest\View(serializerGroups={"Default"})
+     *
+     * @param Book $book
+     * @return View
+     */
+    public function getBook(Book $book): View
+    {
+        return $this->view($book);
+    }
+
+    /**
+     * Creates a Book resource
+     *
+     * @Rest\Post(path="/api/books")
+     * @param Request $request
+     * @return View
+     */
+    public function postBook(Request $request): View
+    {
+        $form = $this->bookManager->postBook($request);
```

Comment:
I would have never expected to receive an instance of Form when I call this method.
If you're doing this for validation, then I'd suggest putting the input into some sort of DTO and validate this DTO directly, without the form.